backend/regalloc/regalloc_split.ml

```diff
@@ -132,48 +133,79 @@ let make_spill :
     ~id:(State.get_and_incr_instruction_id state)
     ~copy ~from:new_reg ~to_:stack_reg
 
-(* Inserts the spills in a block, as early as possible (i.e. immediately after
-   the register is last set), to reduce live ranges. *)
-let rec insert_spills_in_block :
+let rec insert_splills_or_reloads_in_block :
     State.t ->
+    make_spill_or_reload:'a make_operation ->
+    array_of_interest:(Instruction.t -> Reg.t array) ->
+    insert:(Instruction.t DLL.cell -> Instruction.t -> unit) ->
+    add_default:(Instruction.t DLL.t -> Instruction.t -> unit) ->
+    move_cell:(Instruction.t DLL.cell -> Instruction.t DLL.cell option) ->
     block_subst:Substitution.t ->
     stack_subst:Substitution.t ->
     Cfg.basic_block ->
     Instruction.t DLL.cell option ->
     Reg.Set.t ->
     unit =
- fun state ~block_subst ~stack_subst block cell live_at_destruction_point ->
-  match Reg.Set.is_empty live_at_destruction_point with
+ fun state ~make_spill_or_reload ~array_of_interest ~insert ~add_default
+     ~move_cell ~block_subst ~stack_subst block cell live_at_interesting_point ->
+  match Reg.Set.is_empty live_at_interesting_point with
   | true -> ()
   | false -> (
     match cell with
     | None ->
       Reg.Set.iter
         (fun old_reg ->
           let new_reg = Substitution.apply_reg block_subst old_reg in
-          let copy = block.terminator in
-          let spill = make_spill state ~stack_subst ~old_reg ~new_reg ~copy in
-          DLL.add_begin block.body spill)
-        live_at_destruction_point
+          let copy =
+            { Instruction.dummy with
+              dbg = block.terminator.dbg;
+              fdo = block.terminator.fdo;
+              live = block.terminator.live;
+              stack_offset = block.terminator.stack_offset
+            }
```

Comment:
Should this also be a parameter, i.e., first instruction for spill and terminator for reload? In particular for `live` field. Also, it would be nice to avoid the `Instruction.dummy` here.